pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/TransformBlockValSet.java

```diff
@@ -40,12 +44,40 @@
 public class TransformBlockValSet implements BlockValSet {
   private final ProjectionBlock _projectionBlock;
   private final TransformFunction _transformFunction;
+  private final RoaringBitmap _nullBitmap;
 
   private int[] _numMVEntries;
 
-  public TransformBlockValSet(ProjectionBlock projectionBlock, TransformFunction transformFunction) {
+  public TransformBlockValSet(ProjectionBlock projectionBlock, TransformFunction transformFunction,
+      ExpressionContext expression) {
     _projectionBlock = projectionBlock;
     _transformFunction = transformFunction;
+    RoaringBitmap nullBitmap = null;
```

Comment:
I went this way because it requires much less plumbing. Let me know if you feel strongly otherwise.
Also, passing nullHandlingEnabled requires passing QueryContext across many layers. Happy to do so if you prefer.

Reply:
I moved the logic to `getNullBitmap()` method. Please note that the `getNullBitmap()` method is called only when nullHandlingEnabled is set to true.